src/libraries/Native/Unix/System.Native/pal_time.c

```diff
@@ -33,7 +33,15 @@ int32_t SystemNative_UTimensat(const char* path, TimeSpec* times)
 
     updatedTimes[1].tv_sec = (time_t)times[1].tv_sec;
     updatedTimes[1].tv_nsec = (long)times[1].tv_nsec;
-    while (CheckInterrupted(result = utimensat(AT_FDCWD, path, updatedTimes, 0)));
+    while (CheckInterrupted(result = utimensat(AT_FDCWD, path, updatedTimes, AT_SYMLINK_NOFOLLOW)));
+#elif HAVE_LUTIMES
+    struct timeval updatedTimes[2];
+    updatedTimes[0].tv_sec = (long)times[0].tv_sec;
+    updatedTimes[0].tv_usec = (int)times[0].tv_nsec / 1000;
+
+    updatedTimes[1].tv_sec = (long)times[1].tv_sec;
+    updatedTimes[1].tv_usec = (int)times[1].tv_nsec / 1000;
+    while (CheckInterrupted(result = lutimes(path, updatedTimes)));
```

Comment:
You can re-use this code for lutimes and utimes i.e: the `#elif` can be shrinked to lutimes vs utimes.

Reply:
I've combined. Let me know if you want it combined differently (ie. #if ing a whole line rather than a function name).